File: ADTRI.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long int ll;

ll lis[5000005],x;

bool vis[5000004];	

void mark(bool vis[],ll k,ll n)
{
	ll i=2;
	ll num = i*k;

	if(k%4==1)
		lis[k]=1;

	while((num=i*k)<=n)
	{
		vis[num-1]=1;
		if(k%4==1)
			lis[num]=1;
		i++;
	}

}

void seive(ll n)
{
	if (n>=2)
	{
//		memset(vis,0,sizeof(vis));

		for (ll i=1;i<n;i++)
			if (not vis[i])
				mark(vis,i+1,n);
	}
}


int main()
{
	ios::sync_with_stdio(0);
	/*
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);
	*/
	seive(1000004);

	/*for (ll i =1;i<100;i++)
		cout<<lis[i]<<"\n";
	*/ll t;
		scanf("%lld",&t);
	/*cin>>t;*/

	while(t--)
	{
		ll n;
		scanf("%lld",&n);
		/*cin>>n;*/

		if (lis[n] or n==2)
			printf("YES\n");
		else
			printf("NO\n");
	}
}

File: ADTRI.py
import math

lis = [0]*5000007

def markMultiples(vis,k):
	i = 2
	num=i*k
	global lis

	if k%4==1:
		lis[k]=1

	while num <= len(vis):
		vis[num-1] = k

		if k%4==1:
			lis[num]=1

		i+=1
		num=i*k

def SeiveOfEratosthenes(n):
	global lis

	if(n>1):
		vis=[0]*n

		for i in xrange(1,n):
			if not vis[i]:
				markMultiples(vis,i+1)

	return


def main():

	t = int(raw_input())
	SeiveOfEratosthenes(5000005)

	for _ in xrange(t):
		n = int(raw_input())

		if n==2 or lis[n]:
			print "YES"
		else:
			print "NO"
		
main()